### ISSUE TITLE
[A11y] Desktop/Mobile Collection Landing: UO Text Link

### ISSUE THREAD

**Issue by azle** [OregonDigital/OD2#1749]
### Operating System

MacOS

### Browser

Chrome

### Affected page

Collection Show

### Admin user?

- [ ] Yes

### Issue Description

Collection Landing: https://staging.oregondigital.org/collections/gwilliams?locale=en

The institution text link is missing a description for the screen reader.

### Suggested Fix

Needs a descriptive aria label for where the link takes them.

Note:
Clicking the text of "Oregon State University" should run a search across all collections for that same URI, i.e. all OSU items should be returned.

**Comment by CGillen** [comment]
I believe @jechols mentioned this in the webaim list (https://webaim.org/discussion/mail_thread?thread=10202) that this is an iffy failure. 
```
The confusion seems to me to be there no matter how capable a person is (I personally didn't realize what the links did until I followed one, and I'm supposed to already know this)
```
This is probably more of a usability issue than an a11y one. If the text and intent is changed by POSM then we need to revisit this. I think we should work under the theory here that the site only needs to be as accessible as it is usable
